### ISSUE TITLE
CI Lint Checks are redundant

### ISSUE THREAD

**Issue by freddyaboulton** [alteryx/evalml#2527]
Black has a `-t` flag that is used for the target version of python to check against. We currently run two CI checks for linting, one that installs python 3.7 and one that installs python 3.8 with the assumption that installing black with python 3.8 means that it will check against python 3.8.

That's not the case! Our code is actually not formatted for python 3.7/3.8! I am currently running python 3.8.10, when I run black and tell it to check against version 3.8, 22 files would change!

```
black -t py38 --check evalml/
would reformat evalml/pipelines/components/estimators/classifiers/decision_tree_classifier.py
would reformat evalml/pipelines/components/estimators/classifiers/et_classifier.py
would reformat evalml/pipelines/components/estimators/classifiers/elasticnet_classifier.py
would reformat evalml/pipelines/components/estimators/classifiers/kneighbors_classifier.py
would reformat evalml/pipelines/components/estimators/classifiers/logistic_regression_classifier.py
would reformat evalml/pipelines/components/estimators/regressors/decision_tree_regressor.py
would reformat evalml/pipelines/components/estimators/regressors/elasticnet_regressor.py
would reformat evalml/pipelines/components/estimators/classifiers/svm_classifier.py
would reformat evalml/pipelines/components/estimators/regressors/et_regressor.py
would reformat evalml/pipelines/components/transformers/feature_selection/rf_classifier_feature_selector.py
would reformat evalml/pipelines/components/estimators/classifiers/xgboost_classifier.py
would reformat evalml/pipelines/components/estimators/classifiers/catboost_classifier.py
would reformat evalml/pipelines/components/estimators/estimator.py
would reformat evalml/pipelines/components/transformers/feature_selection/rf_regressor_feature_selector.py
would reformat evalml/pipelines/components/estimators/regressors/catboost_regressor.py
would reformat evalml/pipelines/components/estimators/regressors/xgboost_regressor.py
would reformat evalml/pipelines/components/transformers/samplers/undersampler.py
would reformat evalml/pipelines/components/transformers/encoders/target_encoder.py
would reformat evalml/pipelines/components/transformers/samplers/oversamplers.py
would reformat evalml/pipelines/components/transformers/imputers/per_column_imputer.py
would reformat evalml/pipelines/components/estimators/regressors/lightgbm_regressor.py
would reformat evalml/pipelines/components/estimators/classifiers/lightgbm_classifier.py
Oh no! 💥 💔 💥
22 files would be reformatted, 279 files would be left unchanged.
```

Our code is actually currently formatted against python version 3.3:

```
black -t py33 --check evalml/
All done! ✨ 🍰 ✨
301 files would be left unchanged.
```

Given this I think we should 
1. Delete one of the lint jobs
2. Run the remaining lint job with python 3.9 (the latest cpython runtime we support).  This is what one of the package maintainers recommended [here](https://github.com/psf/black/issues/2383). If we explicitly want to check against a python version, we should specify it with the `-t` flag.

**Comment by dsherry** [comment]
@jeremyliweishih says he's running into trouble running `make lint` locally -- it produces different results than on CI. We should fix this.

**Comment by freddyaboulton** [comment]
Closed by #2586
